streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -125,7 +133,10 @@ public void register(StateStore store, boolean loggingEnabled, StateRestoreCallb
         String topic;
         if (loggingEnabled)
             topic = storeChangelogTopic(this.applicationId, store.name());
-        else topic = store.name();
+        else if (sourceStoreToSourceTopic != null && sourceStoreToSourceTopic.containsKey(store.name()))
```

Comment:
My preference is to always use `{..}` for all `if`, `while`, `for` etc statements. It reduces bugs - remember the `goto fail` bug ?